pkg/provider/response.go

```diff
@@ -168,10 +133,11 @@ func (r *Response) makeFailedResponse(
 func (r *Response) makeSuccessfulResponse(
 	attributes *Attributes,
 	timeFormat string,
+	expiration time.Duration,
 ) *samlp.ResponseType {
 	now := time.Now().UTC()
 	nowStr := now.Format(timeFormat)
-	fiveFromNowStr := now.Add(5 * time.Minute).Format(timeFormat)
+	fiveFromNowStr := now.Add(expiration).Format(timeFormat)
```

Comment:
fiveFromNowStr -> validity?

Reply:
Renamed and refactored the usage of the expiration, also in the attributeQuery.